### PR TITLE
ci(.spellcheck.yaml): Update .spellcheck.yaml ReST disable codeblock

### PR DESCRIPTION
**Pull request by neon60**

### CODE REVIEW

**Comment by neon60** [comment]
@MathiasMagnus @saadrahim this solved locally the code block check related issues.

File: .spellcheck.yaml [hunk 1]
```diff
@@ -1,8 +1,37 @@
+# BSD 3-Clause License
+#
+# Copyright (c) 2017-2022, Pytorch contributors
+# All rights reserved.
```
Comment (saadrahim) [inline .spellcheck.yaml:4]: Modifications copyright AMD

File: .spellcheck.yaml [hunk 2]
```diff
@@ -67,3 +83,65 @@ matrix:
         content: '[^\n]+'
         close: ')=$'
   - pyspelling.filters.url:
+- name: reST
+  sources:
+  - 'docs/**/*.rst'
+  dictionary:
+    wordlists:
+      - en-wordlist.txt
```
Comment (saadrahim) [inline .spellcheck.yaml]: file name

File: .spellcheck.yaml [hunk 3]
```diff
@@ -1,8 +1,39 @@
+# BSD 3-Clause License
+#
+# Copyright (c) 2017-2022, Pytorch contributors
+# All rights reserved.
+# Modifications Copyright (c) 2017-2023, Advanced Micro Devices, Inc.
```
Comment (saadrahim) [inline .spellcheck.yaml]:
```suggestion
# Modifications Copyright (c) 2023, Advanced Micro Devices, Inc. 
```

**Review by saadrahim — APPROVED** [review]

**Comment by neon60** [comment]
@saadrahim CI passed finally.

**Review by samjwu — APPROVED** [review]

**Pull request merged**